x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/Waterfall/waterfall_item.tsx

```diff
@@ -216,27 +212,52 @@ export function WaterfallItem({
         </SpanActionToolTip>
         <HttpStatusCode item={item} />
         <NameLabel item={item} />
-        {errorCount > 0 && item.docType === 'transaction' ? (
-          <ErrorOverviewLink
-            serviceName={item.doc.service.name}
-            query={{
-              kuery: `${TRACE_ID} : "${item.doc.trace.id}" and transaction.id : "${item.doc.transaction.id}"`,
-            }}
-            color="danger"
-            style={{ textDecoration: 'none' }}
-          >
-            <EuiToolTip content={tooltipContent}>
-              <ErrorCount count={errorCount} />
-            </EuiToolTip>
-          </ErrorOverviewLink>
-        ) : null}
+
         <Duration item={item} />
+        <RelatedErrors item={item} errorCount={errorCount} />
         {item.docType === 'span' && <SyncBadge sync={item.doc.span.sync} />}
       </ItemText>
     </Container>
   );
 }
 
+function RelatedErrors({
+  item,
+  errorCount,
+}: {
+  item: IWaterfallSpanOrTransaction;
+  errorCount: number;
+}) {
+  const apmRouter = useApmRouter();
+  const theme = useTheme();
+  const { query } = useApmParams('/services/:serviceName/transactions/view');
+
+  const href = apmRouter.link(`/services/:serviceName/errors`, {
+    path: { serviceName: item.doc.service.name },
+    query: {
+      ...query,
+      kuery: `${TRACE_ID} : "${item.doc.trace.id}" and ${TRANSACTION_ID} : "${item.doc.transaction?.id}"`,
+    },
+  });
+
+  if (errorCount > 0) {
+    return (
+      // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+      <div onClick={(e: React.MouseEvent) => e.stopPropagation()}>
+        <EuiBadge href={href} color={theme.eui.euiColorDanger}>
```

Comment:
```suggestion
        <EuiBadge href={href} color={theme.eui.euiColorDanger} iconType="arrowRight">
```

I suggest that we add another visual differentiator to this badge because it's a link and not just a display.

<img width="521" alt="CleanShot 2021-09-08 at 10 30 50@2x" src="https://user-images.githubusercontent.com/4104278/132475033-07aff2bf-5278-4848-83a3-3997699f8434.png">